src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -573,7 +572,7 @@ public Boolean call() throws Exception
 
             // failed to get connection from db, sleep and retry
             quietlySleep(sleepBackoff);
-            sleepBackoff = Math.min(connectionTimeout / 2, (long) (sleepBackoff * 1.3));
+            sleepBackoff = Math.min( TimeUnit.SECONDS.toMillis(5), Math.min(connectionTimeout, (long) (sleepBackoff * 1.3)));
```

Comment:
yes. as I said before: it is minimum of 5 sec or connectionTimeout or sleepBackoff * 1.3

it sleeps for sleepBackoff time and when sleepBackoff becomes more than connectionTimeout, it sleeps for 5 second

that makes it good for smaller connectionTimeout as you wanted for high traffic / 1sec timeout.
also better for larger connectionTimeout. makes HikariCP responsive within 5-10 seconds